crates/store/re_mcap/src/parsers/ros2msg/geometry_msgs/pose_stamped.rs

```diff
@@ -0,0 +1,93 @@
+use super::super::definitions::geometry_msgs::PoseStamped;
+use re_chunk::{Chunk, ChunkId};
+
+use re_sdk_types::archetypes::{CoordinateFrame, InstancePoses3D};
+use re_sdk_types::components::{RotationQuat, Translation3D};
+use re_sdk_types::datatypes::Quaternion;
+
+use super::super::Ros2MessageParser;
+use crate::parsers::{
+    cdr,
+    decode::{MessageParser, ParserContext},
+};
+use crate::util::TimestampCell;
+
+pub struct PoseStampedMessageParser {
+    translations: Vec<Translation3D>,
+    quaternions: Vec<RotationQuat>,
+    frame_ids: Vec<String>,
+}
+
+impl Ros2MessageParser for PoseStampedMessageParser {
+    fn new(num_rows: usize) -> Self {
+        Self {
+            translations: Vec::with_capacity(num_rows),
+            quaternions: Vec::with_capacity(num_rows),
+            frame_ids: Vec::with_capacity(num_rows),
+        }
+    }
+}
+
+impl MessageParser for PoseStampedMessageParser {
+    fn append(&mut self, ctx: &mut ParserContext, msg: &mcap::Message<'_>) -> anyhow::Result<()> {
+        re_tracing::profile_function!();
+        let PoseStamped { header, pose } = cdr::try_decode_message::<PoseStamped>(&msg.data)?;
```

Comment:
Using the destruction pattern all the way ensures we don't ever forget any field. It is a good habit to get into.

```suggestion
        let PoseStamped { header, pose } = cdr::try_decode_message::<PoseStamped>(&msg.data)?;
        let Pose { position, orientation } = pose;
```